lib/tasks/book2.rake

```diff
@@ -83,12 +83,12 @@ def genbook(code, &get_content)
   # revert internal links decorations for ebooks
   content.gsub!(/<<.*?\#(.*?)>>/, "<<\\1>>")
 
-  asciidoc = Asciidoctor::Document.new(content, template_dir: template_dir, attributes: { "lang" => code})
+  asciidoc = Asciidoctor::Document.new(content, attributes: { "lang" => code})
```

Comment:
Hmm. I think the root of the issue is that in newer versions of rails, the earlier https://github.com/git/git-scm.com/blob/7987e258ab41af794cc9474e8637a9b975fc68f7/lib/tasks/book2.rake#L34 is now returning a `Pathname`, whereas asciidoctor is expecting a string. Adding `.to_s` to that line fixes it, as well.

However, I think the `template_dir` parameter is doing nothing, since we stopped using templates in 771e19ad0cf6fe08b3120968f0c15e28349de303. So deleting it makes sense, but we should probably get rid of the `template_dir` variable entirely, since nobody is looking at it.